integration-libs/cds/src/merchandising/facade/cds-merchandising-user-context.service.ts

```diff
@@ -65,7 +72,17 @@ export class CdsMerchandisingUserContextService {
   }
 
   private getConsentReferenceContext(): Observable<MerchandisingUserContext> {
+    let operators: MonoTypeOperatorFunction<ConsentChangedPushEvent>[] = [];
+    if (!!environment.b2b) {
```

Comment:
We shouldn't use the `environment` in the libs. For more, see comment https://github.com/SAP/spartacus/pull/14647/files#r771495613

Reply:
Did you try checking the `channel` property of the `BaseSite` model returned from the method `BaseSiteService.get()`?

Reply:
Ideally, I'd love if we could avoid depending on the info from the `BaseSiteService` and it would be great if you wouldn't need the conditional `startWith` ;) Just to double check: Did you explore alternative approaches to achieve your business goal, that don't depend on reading the `BaseSite` data?

Additional note on performance: 
The method `.get()` will return immediately in case when the data have been loaded before (it is the case for example when the [automatic basesite detection by URL was enabled](https://sap.github.io/spartacus-docs/automatic-context-configuration/).
But will return with a delay (because of the roundtrip to the backend for data), when the method was not accessed before.

Therefore, if you discover performance slowdown when depending on the `.get()` method, you might consider triggering the load of data earlier, if only you find it justified.

Reply:
@znikola helped me realize that currently by default the call for `BaseSite` model won't return the `channel` property. To make it work, we need to add `channel` it in the `fields?` in the configuration of the base site endpoint.
https://github.com/SAP/spartacus/blob/807f6ee39fb7f9455b17c167be9683adf5483694/projects/core/src/occ/adapters/site-context/default-occ-site-context-config.ts#L13

--

But again, I'd love to avoid depending in CDS library on the info about b2b/b2c. Let's discuss first if it's definitely needed

cc @znikola @bgambocjaviniar